pulp_puppet_plugins/pulp_puppet/plugins/importers/directory.py

```diff
@@ -247,18 +248,20 @@ def _import_modules(self, module_paths):
 
         remote_unit_keys = []
 
+        list_of_modules = []
         for module_path in module_paths:
-            if self.canceled:
-                return
             puppet_manifest = self._extract_metadata(module_path)
             module = Module.from_metadata(puppet_manifest)
             remote_unit_keys.append(module.unit_key_str)
+            list_of_modules.append(module)
+
+        pub_step = publish_step.GetLocalUnitsStep(importer_type=constants.IMPORTER_TYPE_ID, available_units=list_of_modules, repo=self.repo)
```

Comment:
It looks like the __init__ method expects importer_type as a positional arg instead of a kwarg[0]

[0] https://github.com/pulp/pulp/blob/master/server/pulp/plugins/util/publish_step.py#L1182